src/plugins/vis_type_vega/public/vega_view/vega_map_view/utils/validataion_helper.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * and the Server Side Public License, v 1; you may not use this file except in
+ * compliance with, at your election, the Elastic License or the Server Side
+ * Public License, v 1.
+ */
+
+import { i18n } from '@kbn/i18n';
+
+export function validateZoomSettings(
+  mapConfig: {
+    maxZoom: number;
+    minZoom: number;
+    zoom?: number;
+  },
+  limits: {
+    maxZoom: number;
+    minZoom: number;
+  },
+  onWarn: (message: any) => void
+) {
+  const DEFAULT_ZOOM = 3;
+
+  let { maxZoom, minZoom, zoom = DEFAULT_ZOOM } = mapConfig;
+
+  const validate = (
```

Comment:
nit: can you change the signature by adding `onWarn`, and just move outside of this closure as a helper function?

```
function validate(
  name: string,
  value: number,
  defaultValue: number,
  min: number,
  max: number,
  onWarn: (message:string)
) {
   ....
}
```